packages/react-components/src/components/AttachmentCard.tsx

```diff
@@ -32,34 +32,29 @@ import { useAttachmentCardStyles, fileNameContainerClassName } from './styles/At
 
 /**
  * @internal
- * _AttachmentCard Component Props.
+ * AttachmentCard Component Props.
  */
 export interface _AttachmentCardProps {
   /**
-   * Attachment name.
+   * File.
    */
-  attachmentName: string;
+  attachment: AttachmentMetadata;
   /**
-   * Extension of the attachment used for rendering the attachment icon.
-   */
-  attachmentExtension: string;
-  /**
-   * Attachment upload progress percentage between 0 and 1.
-   * Attachment transfer progress indicator is only shown when the value is greater than 0 and less than 1.
+   * Optional property to indicate progress of file upload.
    */
   progress?: number;
   /**
-   * Icon to display for actions like download, upload, etc. along the attachment name.
+   * An array of menu actions to be displayed in the attachment card.
    */
-  actionIcon?: JSX.Element;
+  menuActions: AttachmentMenuAction[];
   /**
-   * Function that runs when actionIcon is clicked
+   * Optional arialabel strings for file cards
    */
-  actionHandler?: () => void;
+  strings?: _AttachmentUploadCardsStrings;
   /**
-   * Optional arialabel strings for attachment cards
+   * Optional callback that runs if menu bar action onclick throws.
    */
-  strings?: _AttachmentUploadCardsStrings;
+  onDownloadErrorMessage?: (errMsg: string) => void;
```

Comment:
Rename to handleOnClickError? since AttachmentCard can be for both upload and download

Reply:
i am not sure this error is used for download only. for upload it has a different path which is not related to attachment card (but in send box component instead)

Reply:
Either the comment or the API name needs changing.
I think my preference would be `onActionHandlerFailed(failureMessage: string)`

Reply:
this is an internal component, also i feel like the naming currently is reflecting that it's being used for download only?

`onActionHandlerFailed` might be confusing with upload?

Reply:
As you said, it is being used for downloading but the function is called when the action fails.

Reply:
i am confused, what i am saying is that this property is only being used for download only so the name should reflect download rather than "action" which seems vague to me.

that being said this is just an internal component and not part of public api, so i just updated it